com.unity.render-pipelines.universal/ShaderLibrary/Shadows.hlsl

```diff
@@ -288,6 +288,17 @@ float3 ApplyShadowBias(float3 positionWS, float3 normalWS, float3 lightDirection
     return positionWS;
 }
 
+float ApplyShadowFade(float shadowAttenuation, float3 positionWS)
+{
+    float3 camToPixel = positionWS - _WorldSpaceCameraPos;
+    float distanceCamToPixel2 = dot(camToPixel, camToPixel);
+    float shadowDist = _MainLightShadowParams.z;
+    float startShadowFadeDist = shadowDist * 0.9;
+
+    float fade = saturate((distanceCamToPixel2 - startShadowFadeDist) / (shadowDist - startShadowFadeDist));
+    return shadowAttenuation + (1 - shadowAttenuation) * fade;
```

Comment:
quadratic falloff here should give more natural results.

Reply:
I have tried to implement it in the new version for a faster falloff